task3/UI/script_task7.js

```diff
@@ -84,9 +84,8 @@ class AdCollection {
                     }
                 }
             )
-            if (AdCollection._validate(mutablePost) && this.get(adItem.id)) {
-                this.remove(id);
-                this.add(mutablePost);
+            if (AdCollection._validate(mutablePost) && this.get(id)) {
+                this._adList.splice(id, mutablePost);
```

Comment:
Please take a look at `splice` docs: first argument is an index, second is amount and third is the new item to insert 🙂